ddtrace/llmobs/_llmobs.py

```diff
@@ -272,78 +276,104 @@ def annotate(
         """
-        Tags input/output for a given LLMObs span.
-        io_type: "input" or "output".
-        meta: Span's meta dictionary.
-        data: String or dictionary of key-value pairs to tag as the input or output.
+        if input_messages is not None:
+            if not isinstance(input_messages, Messages):
+                input_messages = Messages(input_messages)
+            try:
+                if input_messages.messages:
+                    span.set_tag_str(INPUT_MESSAGES, json.dumps(input_messages.messages))
+            except (TypeError, AttributeError):
+                log.warning("Failed to parse input messages.")
+        if output_messages is not None:
+            if not isinstance(output_messages, Messages):
+                output_messages = Messages(output_messages)
+            try:
+                if output_messages.messages:
+                    span.set_tag_str(OUTPUT_MESSAGES, json.dumps(output_messages.messages))
+            except (TypeError, AttributeError):
+                log.warning("Failed to parse output messages.")
+
+    @classmethod
+    def _tag_text_io(cls, span, input_value=None, output_value=None):
+        """Tags input/output values for non-LLM kind spans.
+        Will be mapped to span's `meta.{input,output}.values` fields.
         """
-        if io_type not in ("input", "output"):
-            raise ValueError("io_type must be either 'input' or 'output'.")
-        if not isinstance(data, (str, list)):
-            log.warning(
-                "Invalid type, must be either a raw string or list of dictionaries with the format {'content': '...'}."
-            )
-            return
-        if isinstance(data, list) and not all(isinstance(item, dict) for item in data):
-            log.warning("Invalid list item type, must be a list of dictionaries with the format {'content': '...'}.")
-            return
-        span_kind = span.get_tag(SPAN_KIND)
-        tags = None
-        if not span_kind:
-            log.warning("Cannot tag input/output without a span.kind tag.")
-            return
-        if span_kind in ("llm", "agent"):
-            if isinstance(data, str):
-                tags = [{"content": data}]
-            elif isinstance(data, list) and data and isinstance(data[0], dict):
-                tags = data
-            if io_type == "input":
-                span.set_tag_str(INPUT_MESSAGES, json.dumps(tags))
-            else:
-                span.set_tag_str(OUTPUT_MESSAGES, json.dumps(tags))
-        else:
-            if isinstance(data, str):
-                if io_type == "input":
-                    span.set_tag_str(INPUT_VALUE, data)
-                else:
-                    span.set_tag_str(OUTPUT_VALUE, data)
-            else:
-                log.warning("Invalid input/output type for non-llm span. Must be a raw string.")
+        if input_value is not None:
+            span.set_tag_str(INPUT_VALUE, str(input_value))
```

Comment:
I think the frontend only picks up the JSON rendering if we do
`json.dumps(input_value)`

So we might need to do a check like this, something like
```
if isinstance(input_value, str):
      span.set_tag_str(INPUT_VALUE, input_value)
else:
     span.set_tag_str(INPUT_VALUE, json.dumps(input_value))
```